File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.py
class Solution(object):
    def maxAscendingSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        t = nums[0]
        m = 0
        for i in range(1, n):
            if nums[i] > nums[i - 1]:
                t += nums[i]
            else:
                m = max(m, t)
                t = nums[i]
        return max(m, t)

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int n = nums.size();
        int m = 0, t = nums[0];
        for (int i = 1; i < n; ++i) {
            if (nums[i] > nums[i - 1])
                t += nums[i];
            else {
                m = fmax(m, t);
                t = nums[i];
            }
        }
        return fmax(t, m);
    }
};